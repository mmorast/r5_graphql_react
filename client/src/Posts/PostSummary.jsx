import React from 'react'
import PropTypes from 'prop-types'
import {Row, Col} from 'react-bootstrap'
import styled from 'styled-components'
import moment from 'moment'

const PostSummary = props => {
  const {post} = props
  return (
    <PostSummaryArticle>
      <Row>
        <Col sm={12}>
          <PostSummaryTitle>{post.title}</PostSummaryTitle>
          <PostSummaryPublished>{moment(post.publishedAt).format("Do MMMM YYYY")}</PostSummaryPublished>
          <PostSummaryExcerpt>{post.excerpt}</PostSummaryExcerpt>
        </Col>
      </Row>
    </PostSummaryArticle>
  )
}

PostSummary.propTypes = {
  post: PropTypes.object.isRequired,
}

export default PostSummary

const PostSummaryArticle = styled.article``

const PostSummaryTitle = styled.h2``

const PostSummaryPublished = styled.p`
  font-size: 0.8em;
  color: #aaa;
`

const PostSummaryExcerpt = styled.p`
  font-style: italic;
`
