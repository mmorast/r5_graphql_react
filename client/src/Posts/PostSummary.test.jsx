import React from 'react'
import {render} from 'enzyme'
import PostSummary from './PostSummary'

test("Renders PostSummary Component", ()=>{
  const post = {
    id: 1,
    title: "One",
    excerpt: "excerpt one",
    body: "post one body",
    publishedAt: "2017-10-12T12:00:00",
  }

  const actual = render(<PostSummary post={post}/>)
  expect(actual).toMatchSnapshot()
  expect(actual.text()).toMatch(/One/)
  expect(actual.text()).toMatch(/excerpt one/)
})
