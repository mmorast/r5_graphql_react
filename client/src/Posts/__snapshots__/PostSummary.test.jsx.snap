// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders PostSummary Component 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "attribs": Object {
          "class": "sc-bdVaJa bDWFJH",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "row",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "col-sm-12",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "class": "sc-bwzfXH uTxCW",
                    },
                    "children": Array [
                      Object {
                        "data": "One",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "h2",
                    "next": Object {
                      "attribs": Object {
                        "class": "sc-htpNat gqFmdF",
                      },
                      "children": Array [
                        Object {
                          "data": "12th October 2017",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "p",
                      "next": Object {
                        "attribs": Object {
                          "class": "sc-bxivhb fSOxbl",
                        },
                        "children": Array [
                          Object {
                            "data": "excerpt one",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "text",
                          },
                        ],
                        "name": "p",
                        "next": null,
                        "parent": [Circular],
                        "prev": [Circular],
                        "type": "tag",
                      },
                      "parent": [Circular],
                      "prev": [Circular],
                      "type": "tag",
                    },
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                  },
                  Object {
                    "attribs": Object {
                      "class": "sc-htpNat gqFmdF",
                    },
                    "children": Array [
                      Object {
                        "data": "12th October 2017",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "p",
                    "next": Object {
                      "attribs": Object {
                        "class": "sc-bxivhb fSOxbl",
                      },
                      "children": Array [
                        Object {
                          "data": "excerpt one",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "p",
                      "next": null,
                      "parent": [Circular],
                      "prev": [Circular],
                      "type": "tag",
                    },
                    "parent": [Circular],
                    "prev": Object {
                      "attribs": Object {
                        "class": "sc-bwzfXH uTxCW",
                      },
                      "children": Array [
                        Object {
                          "data": "One",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "h2",
                      "next": [Circular],
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                    },
                    "type": "tag",
                  },
                  Object {
                    "attribs": Object {
                      "class": "sc-bxivhb fSOxbl",
                    },
                    "children": Array [
                      Object {
                        "data": "excerpt one",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "p",
                    "next": null,
                    "parent": [Circular],
                    "prev": Object {
                      "attribs": Object {
                        "class": "sc-htpNat gqFmdF",
                      },
                      "children": Array [
                        Object {
                          "data": "12th October 2017",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "p",
                      "next": [Circular],
                      "parent": [Circular],
                      "prev": Object {
                        "attribs": Object {
                          "class": "sc-bwzfXH uTxCW",
                        },
                        "children": Array [
                          Object {
                            "data": "One",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "text",
                          },
                        ],
                        "name": "h2",
                        "next": [Circular],
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                      },
                      "type": "tag",
                    },
                    "type": "tag",
                  },
                ],
                "name": "div",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
              },
            ],
            "name": "div",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
          },
        ],
        "name": "article",
        "next": null,
        "parent": null,
        "prev": null,
        "root": [Circular],
        "type": "tag",
      },
    ],
    "name": "root",
    "next": null,
    "parent": null,
    "prev": null,
    "type": "root",
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xmlMode": false,
  },
}
`;
